WordPress/Classes/ViewRelated/Post/Scheduling/PublishSettingsViewController.swift

```diff
@@ -83,7 +83,11 @@ struct PublishSettingsViewModel {
             } else {
                 post.status = .publish
             }
-        } else {
+        } else if post.originalIsDraft() {
+            // If the original is a draft, keep the post as a draft
+            post.status = .draft
+            post.dateCreated = Date()
+        } else if post.hasFuturePublishDate() {
             post.publishImmediately()
```

Comment:
Im a bit confused by this. Why would we publishImmediately if the post has future publish date?

Reply:
In this case:

1. No date was given (aka the user tapped "Publish immediately"
2. The post is scheduled, which means we're changing it to be published immediately
3. The post is not a draft (it is an already scheduled published)

I'm open to suggestions because it's confusing. A comment might help?

Reply:
A comment would be great!!